### PR TITLE
fix(integrations): Make ApiClient respect REQUESTS_CA_BUNDLE env variable

### LINKED ISSUE #2305: Integrations reject self-signed certificates despite REQUESTS_CA_CERTS set correctly.
### Environment

self-hosted (https://develop.sentry.dev/self-hosted/)

### Steps to Reproduce
Attempt to add an integration to Slack, Opsgenie, or similar integrations that use the shared ApiClient for integrations when running sentry in an environment with self-signed certificates inserted into external requests.
Ensure all certificate routes are correctly placed in `./certificates` as per the guidelines provided in [Sentry Self-Hosted Documentation](https://develop.sentry.dev/self-hosted/custom-ca-roots/).

### Expected Result

The integration should work seamlessly with the self-signed certificate in the certificate chain.

### Actual Result

The integration encounters errors when using prepared requests, as it does not take into account the environment settings. This can lead to SSL certificate verification failures (SSL: CERTIFICATE_VERIFY_FAILED). The use of environment variables, such as REQUESTS_CA_BUNDLE, to modify request behavior is not considered in the prepared request flow.

### Product Area

Settings - Integrations

### Link

_No response_

### DSN

_No response_

### Version

23.6.2

### PR DESCRIPTION
**Pull request by raww**
Fixes: https://github.com/getsentry/self-hosted/issues/2305

When using prepared request flow, the requests library does not take into account the environment. This means that self-signed certificates specified in REQUESTS_CA_BUNDLE will not be taken into account. This change follows the requests documentation guidance of merging the environment settings with your own settings ensuring that they are taken into account. See: [docs](https://requests.readthedocs.io/en/latest/user/advanced/#prepared-requests)

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Comment by raww** [comment]
@hubertdeng123 here it is :)

**Comment by laurikari** [comment]
Thanks @raww!   We just upgraded our on-premise Sentry, and integrations stopped working (any outbound connections go through a proxy with a cert from our private CA).  I'm pretty sure this will fix the issue.

**Comment by laurikari** [comment]
After applying the patch (directly on the running `worker` container) I can confirm that this patch fixes the issue for me.

**Comment by raww** [comment]
@hubertdeng123 looks like the tests won't start year due to:

```please have a collaborator add the [Trigger: getsentry tests] label once they have reviewed the code to trigger downstream tests.
```

**Comment by raww** [comment]
Hi still getting a check failure due to:

```
please have a collaborator add the [Trigger: getsentry tests] label once they have reviewed the code to trigger downstream tests.
```

**Comment by hubertdeng123** [comment]
@getsentry/issues this looks good to me, could I get another set of eyes before merging?

**Comment by raww** [comment]
It would be great to get this merged @hubertdeng123

**Review by hubertdeng123 — APPROVED** [review]

**Comment by laurikari** [comment]
Any chances of somehow nudging this one along @hubertdeng123?

**Comment by raww** [comment]
@hubertdeng123 how can we get this merged?

**Pull request merged**
